internal/jobs/launch_instance_gcp.go

```diff
@@ -50,15 +51,21 @@ func HandleLaunchInstanceGCP(ctx context.Context, job *worker.Job) {
 
 	logger := zerolog.Ctx(ctx).With().Int64("reservation_id", args.ReservationID).Logger()
 	ctx = logger.WithContext(ctx)
+	nc := notifications.GetNotificationClient(ctx)
 
 	jobErr := DoLaunchInstanceGCP(ctx, &args)
 	if jobErr != nil {
 		finishWithError(ctx, args.ReservationID, jobErr)
+		nc.FailedLaunch(ctx, args.ReservationID, jobErr.Error())
 		return
 	}
 
 	jobErr = FetchInstancesDescriptionGCP(ctx, &args)
-
+	if jobErr != nil {
+		nc.FailedLaunch(ctx, args.ReservationID, jobErr.Error())
+	} else {
+		nc.SuccessfulLaunch(ctx, args.ReservationID)
+	}
 	finishJob(ctx, args.ReservationID, jobErr)
 }
```

Comment:
> The function finishWithSuccess is meant to be generic. Currently we only have jobs which launch instances but we might create jobs that do different stuff. I think it might be beneficial to put this code into individual jobs as extra DoSendNotification functions.

finishWithSuccess / finishWithError aren't meant to be generic?

Reply:
True, what you have is better, yeah. They actually take reservationID meaning that they should be only for reservations, but I don’t like this design myself. It is unclear, I could not figure out a better solution.